src/components/List/src/ListItem.tsx

```diff
@@ -1,74 +1,110 @@
 import React from 'react';
 import MaterialListItem from '@material-ui/core/ListItem';
 import BaseProps from '@gemeente-denhaag/baseprops';
-import { ContainerProps } from '@gemeente-denhaag/container';
+import { listitem_classes as classes } from './bem-mapping';
+import { ListItemIcon } from './ListItemIcon';
+import { ListItemSecondaryAction, ListItemText } from '.';
+import { ChevronRightIcon } from '@gemeente-denhaag/icons';
+import IconButton from '@gemeente-denhaag/iconbutton';
 
 export interface ListItemProps extends BaseProps {
-  /**
-   * Defines the `align-items` style property.
-   */
-  alignItems?: 'flex-start' | 'center';
-
   /**
    * If `true`, the list item will be focused during the first mount.
    * Focus will also be triggered if the value changes from false to true.
    */
   autoFocus?: boolean;
 
   /**
-   * If `true`, the list item will be a button (using `ButtonBase`). Props intended
-   * for `ButtonBase` can then be applied to `ListItem`.
-   */
-  button?: false;
-
-  /**
-   * The component used for the root node.
-   * Either a string to use a HTML element or a component.
-   * By default, it's a `li` when `button` is `false` and a `div` when `button` is `true`.
+   * A listitem can be used to navigate, in which case it has a different styling.
+   * Use to specify which type of action this item is.
    */
-  component?: React.ElementType<React.HTMLAttributes<HTMLElement>>;
+  actionType?: 'nav' | 'action';
 
   /**
-   * The container component used when a `ListItemSecondaryAction` is the last child.
+   * The callback fired when clicked
    */
-  ContainerComponent?: React.ElementType<React.HTMLAttributes<HTMLDivElement>>;
+  onClick?: (event: React.MouseEvent) => void;
```

Comment:
I think we want to type this the following way to support touch events: `event: React.MouseEvent | React.TouchEvent`. Check out the ButtonProps for reference.